pkgs/development/lua-modules/generated-packages.nix

```diff
@@ -2416,6 +2416,37 @@ buildLuarocksPackage {
   };
 }) {};
 
+magick = callPackage({ fetchgit, buildLuarocksPackage, lua }:
+buildLuarocksPackage {
+  pname = "magick";
+  version = "1.6.0-1";
+  knownRockspec = (fetchurl {
+    url    = "mirror://luarocks/magick-1.6.0-1.rockspec";
+    sha256 = "1pg150xsxnqvlhxpiy17s9hm4dkc84v46mlwi9rhriynqz8qks9w";
+  }).outPath;
+  src = fetchgit ( removeAttrs (builtins.fromJSON ''{
+  "url": "https://github.com/leafo/magick.git",
+  "rev": "6971fa700c4d392130492a3925344b51c7cc54aa",
+  "date": "2022-03-10T20:02:11-08:00",
+  "path": "/nix/store/fpl99q09zg3qnk4kagxk1djabl1dm47l-magick",
+  "sha256": "01b9qsz27f929rz5z7vapqhazxak74sichdwkjwb219nlhrwfncm",
+  "fetchLFS": false,
+  "fetchSubmodules": true,
+  "deepClone": false,
+  "leaveDotGit": false
+}
+ '') ["date" "path"]) ;
+
+  disabled = (lua.luaversion != "5.1");
```

Comment:
I've tested with `pkgs.lua5_1.withPackages(p: [ p.magick ]);` and ofc it required ffi
➜ result/bin/lua
Lua 5.1.5  Copyright (C) 1994-2012 Lua.org, PUC-Rio
(ins)> require 'magick'
...bd7j-lua-5.1.5-env/share/lua/5.1/magick/wand/lib.lua:1: module 'ffi' not found:
	no field package.preload['ffi']
	no file './share/lua/5.1/ffi.lua'
```

`pkgs.luajit.withPackages(p: [ p.magick ]);` worked so the disabled should check for luajit instead. This can be overriden in overrides.nix

Reply:
Did I do this right? Not sure since `(lua5_1.withPackages(ps: with ps; [ magick ]))` doesn't refuse to evaluate. https://github.com/NixOS/nixpkgs/compare/44216336d88592a8c7c5feb2c2a6fa75f5f4f9b3..6e0f491d657d5f8220537191954c057a5d872789